netlogo-headless/src/main/workspace/AbstractWorkspace.scala

```diff
@@ -187,6 +192,25 @@ with ExtendableWorkspace with ExtensionCompilationEnvironment with APIConformant
   override def lastLogoException: LogoException = null
   override def clearLastLogoException(): Unit = { }
 
+  override lazy val getHubNetManager: Option[HubNetInterface] =
+    Option(Femto.get[HubNetInterface]("org.nlogo.hubnet.server.HeadlessHubNetManager", this,
+                                      FileFormat.standardAnyLoader(true, this),
+                                      FileFormat.converter(getExtensionManager, getLibraryManager,
+                                                           getCompilationEnvironment, this,
+                                                           FileFormat.defaultAutoConvertables)(world.program.dialect)))
+
+  // HubNetWorkspaceInterface
+
+  private var _hubNetRunning = false
+
+  override def hubNetRunning = _hubNetRunning
+
+  override def hubNetRunning_=(running: Boolean): Unit = {
+    _hubNetRunning = running
+  }
+
+  override def getPropertiesInterface: WorldPropertiesInterface =
+    null
```

Comment:
I think the better architecture here would be "favor composition over inheritance", but changing that is probably infeasible, at this point.